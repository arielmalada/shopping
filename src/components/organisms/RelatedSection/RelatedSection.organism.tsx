import React, { useContext } from "react";
import { ProductsByCategoryContext } from "../../../contexts/ProductsByCategoryProvider";
import ProductItem from "../../molecules/ProductItem";

//#region MAIN COMPONENT
const RelatedSection: React.FC = () => {
  const relatedProducts = useContext(ProductsByCategoryContext);
  if (relatedProducts?.products)
    return (
      <section className="flex px-2 pt-10 pb-4 space-x-2 overflow-x-auto">
        {relatedProducts.products.map((item) => (
          <ProductItem
            id={item.id}
            image={item.thumbnail}
            title={item.title}
            price={item.price}
          />
        ))}
      </section>
    );
  return null;
};
//#endregion

export default RelatedSection;
