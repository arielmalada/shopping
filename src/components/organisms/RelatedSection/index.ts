export {default} from './RelatedSection.organism'