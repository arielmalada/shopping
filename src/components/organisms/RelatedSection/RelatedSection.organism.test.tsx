import React from 'react';
import renderer from 'react-test-renderer';
import RelatedSection from './RelatedSection.organism';

describe('Related Section', () => {
  it('should be able to render without crashing', () => {
    const tree = renderer.create(<RelatedSection />).toJSON();

    expect(tree).toMatchSnapshot();
  });
 
});