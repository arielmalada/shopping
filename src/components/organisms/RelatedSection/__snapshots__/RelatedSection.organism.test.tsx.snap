// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Section should be able to render without crashing 1`] = `null`;
